### PR TITLE
feat(code-block): support multiple single-line cmd copy

### PR DESCRIPTION
**Pull request by zchsh**
🎟️ [Asana Task][task]
🔍 [Preview Link][preview]

---

Support copy-button copying of shell-session snippets with multiple single-line commands. 

Support select-copy-paste copying of shell-session snippets by setting `user-select: none` CSS on `.shell-symbol` tokens.

> **Note**: pre-releases has been tested in hashicorp/dev-portal#1149.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

[task]: https://app.asana.com/0/1202097197789424/1203067508838442/f
[preview]: https://react-components-git-zscode-block-multi-command-copy-hashicorp.vercel.app/components/codeblock

### CODE REVIEW

File: packages/code-block/docs.mdx [hunk 1]
```diff
@@ -137,19 +138,61 @@ hello world
...
+| Commands with output                       | ✅        | The command will be copied without the leading `$` or the output.                                                                                                                                  | [🔗](https://learn.hashicorp.com/vault/secrets-management/sm-static-secrets#additional-discussion)                   |
+| Non-shell Snippets                         | ✅        | The entire snippet will be copied.                                                                                                                                                                 | [🔗](https://learn.hashicorp.com/nomad/getting-started/jobs#modifying-a-job)                                         |
+| Single-line commands across multiple lines | ✅        | Commands will be copied without their leading `$ `. Lines that do not start in `$ ` will be omitted from the copied snippet.                                                                       | ⏳ (none yet, relatively new feature)                                                                                |
```
Comment (zchsh) [inline packages/code-block/docs.mdx:194]: Updated this table, we now support many single-line commands in a single `shell-session` snippet 👍

File: packages/code-block/partials/code-lines/style.module.css [hunk 2]
```diff
@@ -12,6 +12,13 @@ pre.pre {
   padding: 0;
   margin: 0;
   background-color: transparent;
+
+  /* Prevent shell-symbols from being selected, for better select-copy-paste */
+  &:global(.language-shell-session) {
+    & :global(.token.shell-symbol) {
+      user-select: none;
```
Comment (zchsh) [inline packages/code-block/partials/code-lines/style.module.css:19]: CSS fix for the "select-copy-paste" use case. This applies to all code-blocks with `$` characters (when correctly identified as `.shell-symbol`, which of course requires highlighting).

File: packages/code-block/utils/process-snippet.js [hunk 3]
```diff
@@ -18,20 +18,51 @@ function processSnippet(snippet) {
...
+   * - Or maybe just need to look into `shellwords` use here?
+   *   (Full disclosure, I didn't set it up, so am not familiar.
+   *   it seems like it could be part of some solution?)
+   *
+   * ref: https://github.com/jimmycuadra/shellwords (really light on docs)
+   * ref: https://ruby-doc.org/stdlib-1.9.3/libdoc/shellwords/rdoc/Shellwords.html
+   *      (ruby module of same name, seems to have better docs)
+   *
+   * Related task:
+   * https://app.asana.com/0/1100423001970639/1199504357822173/f
```
Comment (zchsh) [inline packages/code-block/utils/process-snippet.js:44]: Would love to explore better copying behaviour at some point in the future... but for now, was definitely reaching for a more immediate solution to ensure our "copy these install commands" works on download pages. See hashicorp/dev-portal#1149.

**Comment by zchsh** [comment]
@BRKalow This update to `code-block` fixes things so that on `/downloads` views in Dev Dot, the "install" commands still copy correctly.

Have proven this out in hashicorp/dev-portal#1149. Let me know what you think 🙇‍♂️

**Comment by zchsh** [comment]
@BrandonRomano Tagging ya here in cause you have extra review bandwidth ☺️

File: packages/code-block/utils/process-snippet.js [hunk 4]
```diff
@@ -18,20 +18,51 @@ function processSnippet(snippet) {
...
+   *
+   */
+  const firstLine = lines[0]
+  const isMultiLineCommand =
+    firstLine.endsWith('\\') || firstLine.endsWith('EOF')
+  console.log({ firstLine, isMultiLineCommand })
```
Comment (BRKalow) [inline packages/code-block/utils/process-snippet.js:50]: 👀 
```suggestion
```

**Review by BRKalow — APPROVED** [review]
Nice improvements to the swingset page! I did some testing and all seems to be working. Left a suggestion on removing a console log

File: packages/code-block/utils/process-snippet.js (hunk 4 above)
Comment (zchsh) [reply packages/code-block/utils/process-snippet.js]: oooooops 😬 ty! 💖

**Pull request merged**